### PR TITLE
fix: 'promises must be awaited' sonar issues

### PR DESCRIPTION
**Pull request by kranthie-sap**
# Issue
Sonar Cloud introduced a new TypeScript rule '[Promises should not be misused](https://sonarcloud.io/organizations/sap-1/rules?open=typescript:S6544&rule_key=typescript:S6544)' and this makes our [quality gate](https://sonarcloud.io/project/issues?resolved=false&types=BUG&sinceLeakPeriod=true&severities=BLOCKER,CRITICAL,MAJOR,MINOR&id=SAP_open-ux-tools) fail due to new issues reported.

# Description
* Introduce the corresponding `eslint` rule `@typescript-eslint/no-floating-promises` so developers see the issue locally
* Add `.eslintrc` files in each module to fix the following error when enabling the above `eslint` rule:
```
Error: Error while loading rule '@typescript-eslint/no-floating-promises': You have used a rule which requires parserServices to be generated. You must therefore provide a value for the "parserOptions.project" property for @typescript-eslint/parser.
```
* Add new `tsconfig.esbuild.json` to include `test` sources to fix errors like:
```
packages/yaml/test/yaml-document.test.ts
  0:0  error  Parsing error: ESLint was configured to run on `<tsconfigRootDir>/test/yaml-document.test.ts` using `parserOptions.project`: <tsconfigRootDir>/packages/yaml/tsconfig.json
However, that TSConfig does not include this file. Either:
- Change ESLint's list of included files to not include this file
- Change that TSConfig to include this file
- Create a new TSConfig that includes this file and include it in your parserOptions.project
See the typescript-eslint docs for more info: https://typescript-eslint.io/linting/troubleshooting#i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-file
``` 
* Fix issues reported corresponding to the new rules

### CODE REVIEW

**Review by tobiasqueck — CHANGES_REQUESTED** [review]
Is the `.eslint copy` file in `examples/odata-cli` intended?

Otherwise, the configuration changes look as expected and the code fixes make sense. 
Changeset looks good

**Review by tobiasqueck — CHANGES_REQUESTED** [review]
Is the `.eslint copy` file in `examples/odata-cli` intended?

**Comment by kranthie-sap** [comment]
Thanks @tobiasqueck for catching the redundant file. I removed it.

File: examples/odata-cli/.eslintrc [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
     "extends": ["../../.eslintrc"],
+    "parserOptions": { "project": "./tsconfig.esbuild.json" },
```
Comment (devinea) [inline examples/odata-cli/.eslintrc:3]:
```suggestion
    "parserOptions": { "project": "./tsconfig.eslint.json" },
```
As discussed please rename the config file

Comment (kranthie-sap) [reply examples/odata-cli/.eslintrc]: Thanks, @devinea for catching this typo. I renamed the files.

File: .eslintrc [hunk 2]
```diff
@@ -88,9 +89,11 @@
             "files": ["**/test/**/*.js", "**/test/**/*.ts", "**/test/**/*.tsx"],
             "rules": {
                 "no-console": "off",
+                "jsdoc/require-jsdoc": "off",
                 "jsdoc/require-returns-description": "off",
                 "jsdoc/require-param-description": "off",
-                "max-nested-callbacks": ["warn", 5]
+                "max-nested-callbacks": ["warn", 5],
+                "sonarjs/cognitive-complexity": "off"
```
Comment (tobiasqueck) [inline .eslintrc:96]: How did this one `"sonarjs/cognitive-complexity": "off"` sneak in???

**Review by tobiasqueck — CHANGES_REQUESTED** [review]
I am a big fan of the `"sonarjs/cognitive-complexity"` rule so I am curios why we turn it off.

File: .eslintrc (hunk 2 above)
Comment (kranthie-sap) [reply .eslintrc:96]: This is only for the test files. There were some tests where this was failing and I think it should be OK for this to be turned off for tests.

**Review by devinea — APPROVED** [review]
Thanks @kranthie-sap 
Updates to eslint config , test and source code looks good. 
Changeset included.

**Review by Klaus-Keller — APPROVED** [review]
I focused my review on `cap-config-writer`, `create`, `environment-check`, `mockserver-config-writer`, `project-access`, `ui-components`, `ui5-proxy-middleware`.
- code changes look good and are reasonable
- thanks for fixing some typos
- did not checkout and run locally
- it is a bit unfortunate that we have to add `tsconfig.eslint.json` without meaningful content to each and every module just to get the rules working properly. But I understand that no other solution could be found.  

Approved from my side.

**Review by tobiasqueck — APPROVED** [review]

**Pull request merged**